cmd/tendermint/commands/init.go

```diff
@@ -17,29 +15,34 @@ var InitFilesCmd = &cobra.Command{
 }
 
 func initFiles(cmd *cobra.Command, args []string) {
+	// private validator
 	privValFile := config.PrivValidatorFile()
-	if _, err := os.Stat(privValFile); os.IsNotExist(err) {
-		privValidator := types.GenPrivValidatorFS(privValFile)
+	var privValidator *types.PrivValidatorFS
+	if cmn.FileExists(privValFile) {
+		privValidator = types.LoadPrivValidatorFS(privValFile)
+		logger.Info("Found private validator", "path", privValFile)
+	} else {
+		privValidator = types.GenPrivValidatorFS(privValFile)
 		privValidator.Save()
+		logger.Info("Genetated private validator", "path", privValFile)
+	}
 
-		genFile := config.GenesisFile()
-
-		if _, err := os.Stat(genFile); os.IsNotExist(err) {
-			genDoc := types.GenesisDoc{
-				ChainID: cmn.Fmt("test-chain-%v", cmn.RandStr(6)),
-			}
-			genDoc.Validators = []types.GenesisValidator{{
-				PubKey: privValidator.GetPubKey(),
-				Power:  10,
-			}}
-
-			if err := genDoc.SaveAs(genFile); err != nil {
-				panic(err)
-			}
+	// genesis file
+	genFile := config.GenesisFile()
+	if cmn.FileExists(genFile) {
```

Comment:
note there can be a situation where there is a genesis file, but no private validator. then genesis will not be modified. but I think it's fine.